simulator/modules/decode/decode.cpp

```diff
@@ -48,12 +48,12 @@ Decode<FuncInstr>::Decode( bool log) : Log( log)
 
 
 template <typename FuncInstr>
-typename Decode<FuncInstr>::Instr Decode<FuncInstr>::read_instr( Cycle cycle)
+std::pair<typename Decode<FuncInstr>::Instr, bool> Decode<FuncInstr>::read_instr( Cycle cycle)
```

Comment:
Same here

Reply:
Here I have troubles when I want to use _Instr_ instead of _typename Decode<FuncInstr>::Instr_, but inside of function it is ok.